### PR TITLE
Some path targeting stuff

### PR DESCRIPTION
**Pull request by liach**
I think they are very open to improvements, but I will share what I have for now

Signed-off-by: liach <liach@users.noreply.github.com>

### CODE REVIEW

File: mappings/net/minecraft/entity/ai/FuzzyPositions.mapping [hunk 1]
```diff
@@ -0,0 +1,51 @@
+CLASS net/minecraft/class_5535 net/minecraft/entity/ai/FuzzyPositions
+	METHOD method_31537 towardTarget (Lnet/minecraft/class_1314;ILjava/util/Random;Lnet/minecraft/class_2338;)Lnet/minecraft/class_2338;
+		COMMENT Adjusts the input {@code fuzz} slightly toward the given {@code entity}'s
+		COMMENT {@link net.minecraft.entity.mob.MobEntity$getPositionTarget() position target}
```
Comment (liach) [inline mappings/net/minecraft/entity/ai/FuzzyPositions.mapping]:
```suggestion
		COMMENT {@link net.minecraft.entity.mob.MobEntity#getPositionTarget() position target}
```
lolwhat

**Comment by liach** [comment]
Also, for sweeping unmapped top level classes, #1967 sweeps class_5565 to 5584 and probably deserves some reviews

File: mappings/net/minecraft/entity/ai/AboveGroundTargeting.mapping [hunk 2]
```diff
@@ -0,0 +1,14 @@
+CLASS net/minecraft/class_5533 net/minecraft/entity/ai/AboveGroundTargeting
+	COMMENT Similar to {@link FuzzyTargeting}, but the positions this class' utility methods
+	COMMENT find never have pathfinding penalties and is always above ground or water.
```
Comment (YanisBft) [inline mappings/net/minecraft/entity/ai/AboveGroundTargeting.mapping]:
```suggestion
	COMMENT find never have pathfinding penalties and are always above ground or water.
```

File: mappings/net/minecraft/entity/ai/FuzzyPositions.mapping [hunk 3]
```diff
@@ -0,0 +1,51 @@
...
+		COMMENT and {@code verticalRange} and returns {@code null} if it cannot do so.
+		ARG 0 random
+		ARG 1 horizontalRange
+		ARG 2 verticalRange
+		ARG 3 startHeight
+		ARG 4 xDirection
+		ARG 6 zDirection
+		ARG 8 angleRange
+	METHOD method_31543 guessBest (Ljava/util/function/Supplier;Ljava/util/function/ToDoubleFunction;)Lnet/minecraft/class_243;
+		COMMENT Returns the {@link Vec3d#ofBottomCenter(BlockPos) bottom center} of  a highest scoring
```
Comment (YanisBft) [inline mappings/net/minecraft/entity/ai/FuzzyPositions.mapping]:
```suggestion
		COMMENT Returns the {@link Vec3d#ofBottomCenter(BlockPos) bottom center} of a highest scoring
```

File: mappings/net/minecraft/entity/ai/NavigationConditions.mapping [hunk 4]
```diff
@@ -0,0 +1,25 @@
...
+	METHOD method_31518 isWaterAt (Lnet/minecraft/class_1314;Lnet/minecraft/class_2338;)Z
+		ARG 0 entity
+		ARG 1 pos
+	METHOD method_31519 isInvalidPosition (Lnet/minecraft/class_1408;Lnet/minecraft/class_2338;)Z
+		ARG 0 navigation
+		ARG 1 pos
+	METHOD method_31520 isHeightInvalid (Lnet/minecraft/class_2338;Lnet/minecraft/class_1314;)Z
+		ARG 0 pos
+		ARG 1 entity
+	METHOD method_31521 isPositonTargetOutOfWalkRange (ZLnet/minecraft/class_1314;Lnet/minecraft/class_2338;)Z
```
Comment (YanisBft) [inline mappings/net/minecraft/entity/ai/NavigationConditions.mapping]:
```suggestion
	METHOD method_31521 isPositionTargetOutOfWalkRange (ZLnet/minecraft/class_1314;Lnet/minecraft/class_2338;)Z
```

File: mappings/net/minecraft/entity/ai/NoPenaltySolidTargeting.mapping [hunk 5]
```diff
@@ -0,0 +1,22 @@
+CLASS net/minecraft/class_5530 net/minecraft/entity/ai/NoPenaltySolidTargeting
+	COMMENT Similar to {@link FuzzyTargeting}, but the positions this class' utility methods
+	COMMENT find never have pathfinding penalties and is always on solid.
```
Comment (YanisBft) [inline mappings/net/minecraft/entity/ai/NoPenaltySolidTargeting.mapping]:
```suggestion
	COMMENT find never have pathfinding penalties and are always on solid.
```

**Review by YanisBft — APPROVED** [review]

**Comment by liach** [comment]
Hmm, anyone mind review?

**Review by modmuss50 — APPROVED** [review]
Im far from an expert on the AI stuff, cannot notice anything that seems wrong.

File: mappings/net/minecraft/entity/ai/NoPenaltySolidTargeting.mapping [hunk 6]
```diff
@@ -0,0 +1,22 @@
+CLASS net/minecraft/class_5530 net/minecraft/entity/ai/NoPenaltySolidTargeting
+	COMMENT Similar to {@link FuzzyTargeting}, but the positions this class' utility methods
+	COMMENT find never have pathfinding penalties and are always on solid.
```
Comment (i509VCB) [inline mappings/net/minecraft/entity/ai/NoPenaltySolidTargeting.mapping]: on solid blocks?

Comment (liach) [reply mappings/net/minecraft/entity/ai/NoPenaltySolidTargeting.mapping]:
```suggestion
	COMMENT find never have pathfinding penalties and are always on solid blocks.
```
Sure

**Comment by liach** [comment]
If you guys find problems with this pr, remove `ready to merge`.

**Pull request merged**